internal/gapicgen/generator/gapics.go

```diff
@@ -86,25 +100,85 @@ func (g *GapicGenerator) Regen(ctx context.Context) error {
 		return err
 	}
 
-	if err := g.dropModReplaceGenproto(); err != nil {
+	if err := forEachMod(g.googleCloudDir, g.dropModReplaceGenproto); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// forEachMod runs the given function with the directory of
+// every non-internal module.
+func forEachMod(rootDir string, fn func(dir string) error) error {
+	return filepath.WalkDir(rootDir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if strings.Contains(path, "internal") {
+			return filepath.SkipDir
+		}
+		if d.Name() == "go.mod" {
+			if err := fn(filepath.Dir(path)); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+}
+
+func goModTidy(dir string) error {
+	log.Printf("[%s] running go mod tidy\n", dir)
```

Comment:
Do you want the extra newline? One should be appended already with the log package.

Reply:
Done.